transport/graphsync/graphsync.go

```diff
@@ -981,7 +920,7 @@ func (c *dtChannel) gsDataRequestRcvd(gsKey graphsyncKey, hookActions graphsync.
 
 	// Save a mapping from the graphsync key to the channel ID so that
 	// subsequent graphsync callbacks are associated with this channel
-	c.gsKey = gsKey
+	c.gsKey = &gsKey
```

Comment:
We don't seem to have acquired the lock before writing 🤔

Reply:
The lock is acquired on line 515. The code in this file is super complicated to follow.

Reply:
Ack.